pkg/client/cache/delta_fifo.go

```diff
@@ -452,6 +460,16 @@ func (f *DeltaFIFO) Replace(list []interface{}, resourceVersion string) error {
 	return nil
 }
 
+// Resync will send a sync event for each item
+func (f *DeltaFIFO) Resync() error {
+	f.lock.RLock()
+	defer f.lock.RUnlock()
+	if err := f.replaceLocked(f.listLocked()); err != nil {
```

Comment:
That is what was originally there, but I changed to replace because replace will do deletion detection.  This allows the questionable controller_test.go#TestUpdate unit test to pass, which relies on the deletion detection.  I don't feel that is a realistic scenario in real world operation with these changes though.  IMO, delete or heavily modify the TestUpdate test and go with queueActionLocked impl, but wanted to discuss with you.